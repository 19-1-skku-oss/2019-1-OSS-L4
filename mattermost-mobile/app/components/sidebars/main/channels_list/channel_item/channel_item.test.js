// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';
import {TouchableHighlight} from 'react-native';

import Preferences from 'mattermost-redux/constants/preferences';

import ChannelItem from './channel_item.js';

jest.useFakeTimers();
jest.mock('react-intl');

describe('ChannelItem', () => {
    const channel = {
        id: 'channel_id',
        delete_at: 0,
        type: 'O',
        fake: false,
        status: 'online',
    };

    const baseProps = {
        channelId: 'channel_id',
        channel,
        currentChannelId: 'current_channel_id',
        displayName: 'display_name',
        isChannelMuted: false,
        currentUserId: 'currentUser',
        isUnread: true,
        hasDraft: false,
        mentions: 0,
        navigator: {push: () => {}}, // eslint-disable-line no-empty-function
        onSelectChannel: () => {}, // eslint-disable-line no-empty-function
        shouldHideChannel: false,
        showUnreadForMsgs: true,
        theme: Preferences.THEMES.default,
        unreadMsgs: 1,
        isSearchResult: false,
        isBot: false,
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <ChannelItem {...baseProps}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot with mentions and muted', () => {
        const newProps = {
            ...baseProps,
            mentions: 1,
            isChannelMuted: true,
        };

        const wrapper = shallow(
            <ChannelItem {...newProps}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot for deactivated user and not searchResults or currentChannel', () => {
        const channelObj = {
            ...channel,
            type: 'D',
            delete_at: 123,
        };

        const newProps = {
            ...baseProps,
            channel: channelObj,
        };

        const wrapper = shallow(
            <ChannelItem {...newProps}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot for deactivated user and is searchResult', () => {
        const channelObj = {
            ...channel,
            type: 'D',
            delete_at: 123,
        };

        const newProps = {
            ...baseProps,
            isSearchResult: true,
            channel: channelObj,
        };

        const wrapper = shallow(
            <ChannelItem {...newProps}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot for deactivated user and is currentChannel', () => {
        const channelObj = {
            ...channel,
            type: 'D',
            delete_at: 123,
        };

        const newProps = {
            ...baseProps,
            channel: channelObj,
            currentChannelId: 'channel_id',
        };

        const wrapper = shallow(
            <ChannelItem {...newProps}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot for no displayName', () => {
        const newProps = {
            ...baseProps,
            displayName: '',
        };

        const wrapper = shallow(
            <ChannelItem {...newProps}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot for current user i.e currentUser (you)', () => {
        const channelObj = {
            ...channel,
            type: 'D',
            teammate_id: 'currentUser',
        };

        const newProps = {
            ...baseProps,
            channel: channelObj,
            currentChannelId: 'channel_id',
        };

        const wrapper = shallow(
            <ChannelItem {...newProps}/>,
            {context: {intl: {formatMessage: (intlId) => intlId.defaultMessage}}},
        );
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot for current user i.e currentUser (you) when isSearchResult', () => {
        const channelObj = {
            ...channel,
            id: 'currentUser',
            type: 'D',
            teammate_id: 'somethingElse',
        };

        const newProps = {
            ...baseProps,
            channel: channelObj,
            currentChannelId: 'channel_id',
            isSearchResult: true,
        };

        const wrapper = shallow(
            <ChannelItem {...newProps}/>,
            {context: {intl: {formatMessage: (intlId) => intlId.defaultMessage}}},
        );
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot with draft', () => {
        const wrapper = shallow(
            <ChannelItem
                {...baseProps}
                hasDraft={true}
            />,
            {context: {intl: {formatMessage: jest.fn()}}},
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot for showUnreadForMsgs', () => {
        const wrapper = shallow(
            <ChannelItem
                {...baseProps}
                hasDraft={true}
                shouldHideChannel={true}
                unreadMsgs={0}
            />,
            {context: {intl: {formatMessage: jest.fn()}}},
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('Should call onPress', () => {
        const onSelectChannel = jest.fn();

        const wrapper = shallow(
            <ChannelItem
                {...baseProps}
                onSelectChannel={onSelectChannel}
            />,
            {context: {intl: {formatMessage: jest.fn()}}},
        );

        wrapper.find(TouchableHighlight).simulate('press');
        jest.runAllTimers();

        const expectedChannelParams = {id: baseProps.channelId, display_name: baseProps.displayName, fake: channel.fake, type: channel.type};
        expect(onSelectChannel).toHaveBeenCalledWith(expectedChannelParams, baseProps.currentChannelId);
    });
});
