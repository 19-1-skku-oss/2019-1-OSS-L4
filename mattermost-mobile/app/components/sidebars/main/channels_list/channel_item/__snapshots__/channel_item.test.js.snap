// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelItem should match snapshot 1`] = `
<AnimatedComponent>
  <TouchableHighlight
    activeOpacity={0.85}
    delayPressOut={100}
    onLongPress={[Function]}
    onPress={[Function]}
    underlayColor="rgba(69,120,191,0.5)"
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "row",
            "height": 44,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "paddingLeft": 16,
            },
            undefined,
          ]
        }
      >
        <ChannelIcon
          channelId="channel_id"
          hasDraft={false}
          isActive={false}
          isArchived={false}
          isBot={false}
          isInfo={false}
          isUnread={true}
          membersCount={1}
          size={16}
          status="online"
          theme={
            Object {
              "awayIndicator": "#ffbc42",
              "buttonBg": "#166de0",
              "buttonColor": "#ffffff",
              "centerChannelBg": "#ffffff",
              "centerChannelColor": "#3d3c40",
              "codeTheme": "github",
              "dndIndicator": "#f74343",
              "errorTextColor": "#fd5960",
              "linkColor": "#2389d7",
              "mentionBj": "#ffffff",
              "mentionColor": "#145dbf",
              "mentionHighlightBg": "#ffe577",
              "mentionHighlightLink": "#166de0",
              "newMessageSeparator": "#ff8800",
              "onlineIndicator": "#06d6a0",
              "sidebarBg": "#145dbf",
              "sidebarHeaderBg": "#1153ab",
              "sidebarHeaderTextColor": "#ffffff",
              "sidebarText": "#ffffff",
              "sidebarTextActiveBorder": "#579eff",
              "sidebarTextActiveColor": "#ffffff",
              "sidebarTextHoverBg": "#4578bf",
              "sidebarUnreadText": "#ffffff",
              "type": "Mattermost",
            }
          }
          type="O"
        />
        <Text
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "rgba(255,255,255,0.4)",
                "flex": 1,
                "fontSize": 14,
                "fontWeight": "600",
                "height": "100%",
                "lineHeight": 44,
                "paddingRight": 10,
                "textAlignVertical": "center",
              },
              Object {
                "color": "#ffffff",
              },
            ]
          }
        >
          display_name
        </Text>
      </View>
    </View>
  </TouchableHighlight>
</AnimatedComponent>
`;

exports[`ChannelItem should match snapshot for current user i.e currentUser (you) 1`] = `
<AnimatedComponent>
  <TouchableHighlight
    activeOpacity={0.85}
    delayPressOut={100}
    onLongPress={[Function]}
    onPress={[Function]}
    underlayColor="rgba(69,120,191,0.5)"
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "row",
            "height": 44,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#579eff",
            "width": 5,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "paddingLeft": 16,
            },
            Object {
              "backgroundColor": "rgba(255,255,255,0.1)",
              "paddingLeft": 11,
            },
          ]
        }
      >
        <ChannelIcon
          channelId="channel_id"
          hasDraft={false}
          isActive={true}
          isArchived={false}
          isBot={false}
          isInfo={false}
          isUnread={true}
          membersCount={1}
          size={16}
          status="online"
          theme={
            Object {
              "awayIndicator": "#ffbc42",
              "buttonBg": "#166de0",
              "buttonColor": "#ffffff",
              "centerChannelBg": "#ffffff",
              "centerChannelColor": "#3d3c40",
              "codeTheme": "github",
              "dndIndicator": "#f74343",
              "errorTextColor": "#fd5960",
              "linkColor": "#2389d7",
              "mentionBj": "#ffffff",
              "mentionColor": "#145dbf",
              "mentionHighlightBg": "#ffe577",
              "mentionHighlightLink": "#166de0",
              "newMessageSeparator": "#ff8800",
              "onlineIndicator": "#06d6a0",
              "sidebarBg": "#145dbf",
              "sidebarHeaderBg": "#1153ab",
              "sidebarHeaderTextColor": "#ffffff",
              "sidebarText": "#ffffff",
              "sidebarTextActiveBorder": "#579eff",
              "sidebarTextActiveColor": "#ffffff",
              "sidebarTextHoverBg": "#4578bf",
              "sidebarUnreadText": "#ffffff",
              "type": "Mattermost",
            }
          }
          type="D"
        />
        <Text
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "rgba(255,255,255,0.4)",
                "flex": 1,
                "fontSize": 14,
                "fontWeight": "600",
                "height": "100%",
                "lineHeight": 44,
                "paddingRight": 10,
                "textAlignVertical": "center",
              },
              Object {
                "color": "#ffffff",
              },
            ]
          }
        >
          {displayName} (you)
        </Text>
      </View>
    </View>
  </TouchableHighlight>
</AnimatedComponent>
`;

exports[`ChannelItem should match snapshot for current user i.e currentUser (you) when isSearchResult 1`] = `
<AnimatedComponent>
  <TouchableHighlight
    activeOpacity={0.85}
    delayPressOut={100}
    onLongPress={[Function]}
    onPress={[Function]}
    underlayColor="rgba(69,120,191,0.5)"
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "row",
            "height": 44,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#579eff",
            "width": 5,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "paddingLeft": 16,
            },
            Object {
              "backgroundColor": "rgba(255,255,255,0.1)",
              "paddingLeft": 11,
            },
          ]
        }
      >
        <ChannelIcon
          channelId="channel_id"
          hasDraft={false}
          isActive={true}
          isArchived={false}
          isBot={false}
          isInfo={false}
          isUnread={true}
          membersCount={1}
          size={16}
          status="online"
          theme={
            Object {
              "awayIndicator": "#ffbc42",
              "buttonBg": "#166de0",
              "buttonColor": "#ffffff",
              "centerChannelBg": "#ffffff",
              "centerChannelColor": "#3d3c40",
              "codeTheme": "github",
              "dndIndicator": "#f74343",
              "errorTextColor": "#fd5960",
              "linkColor": "#2389d7",
              "mentionBj": "#ffffff",
              "mentionColor": "#145dbf",
              "mentionHighlightBg": "#ffe577",
              "mentionHighlightLink": "#166de0",
              "newMessageSeparator": "#ff8800",
              "onlineIndicator": "#06d6a0",
              "sidebarBg": "#145dbf",
              "sidebarHeaderBg": "#1153ab",
              "sidebarHeaderTextColor": "#ffffff",
              "sidebarText": "#ffffff",
              "sidebarTextActiveBorder": "#579eff",
              "sidebarTextActiveColor": "#ffffff",
              "sidebarTextHoverBg": "#4578bf",
              "sidebarUnreadText": "#ffffff",
              "type": "Mattermost",
            }
          }
          type="D"
        />
        <Text
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "rgba(255,255,255,0.4)",
                "flex": 1,
                "fontSize": 14,
                "fontWeight": "600",
                "height": "100%",
                "lineHeight": 44,
                "paddingRight": 10,
                "textAlignVertical": "center",
              },
              Object {
                "color": "#ffffff",
              },
            ]
          }
        >
          {displayName} (you)
        </Text>
      </View>
    </View>
  </TouchableHighlight>
</AnimatedComponent>
`;

exports[`ChannelItem should match snapshot for deactivated user and is currentChannel 1`] = `
<AnimatedComponent>
  <TouchableHighlight
    activeOpacity={0.85}
    delayPressOut={100}
    onLongPress={[Function]}
    onPress={[Function]}
    underlayColor="rgba(69,120,191,0.5)"
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "row",
            "height": 44,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#579eff",
            "width": 5,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "paddingLeft": 16,
            },
            Object {
              "backgroundColor": "rgba(255,255,255,0.1)",
              "paddingLeft": 11,
            },
          ]
        }
      >
        <ChannelIcon
          channelId="channel_id"
          hasDraft={false}
          isActive={true}
          isArchived={true}
          isBot={false}
          isInfo={false}
          isUnread={true}
          membersCount={1}
          size={16}
          status="online"
          theme={
            Object {
              "awayIndicator": "#ffbc42",
              "buttonBg": "#166de0",
              "buttonColor": "#ffffff",
              "centerChannelBg": "#ffffff",
              "centerChannelColor": "#3d3c40",
              "codeTheme": "github",
              "dndIndicator": "#f74343",
              "errorTextColor": "#fd5960",
              "linkColor": "#2389d7",
              "mentionBj": "#ffffff",
              "mentionColor": "#145dbf",
              "mentionHighlightBg": "#ffe577",
              "mentionHighlightLink": "#166de0",
              "newMessageSeparator": "#ff8800",
              "onlineIndicator": "#06d6a0",
              "sidebarBg": "#145dbf",
              "sidebarHeaderBg": "#1153ab",
              "sidebarHeaderTextColor": "#ffffff",
              "sidebarText": "#ffffff",
              "sidebarTextActiveBorder": "#579eff",
              "sidebarTextActiveColor": "#ffffff",
              "sidebarTextHoverBg": "#4578bf",
              "sidebarUnreadText": "#ffffff",
              "type": "Mattermost",
            }
          }
          type="D"
        />
        <Text
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "rgba(255,255,255,0.4)",
                "flex": 1,
                "fontSize": 14,
                "fontWeight": "600",
                "height": "100%",
                "lineHeight": 44,
                "paddingRight": 10,
                "textAlignVertical": "center",
              },
              Object {
                "color": "#ffffff",
              },
            ]
          }
        >
          display_name
        </Text>
      </View>
    </View>
  </TouchableHighlight>
</AnimatedComponent>
`;

exports[`ChannelItem should match snapshot for deactivated user and is searchResult 1`] = `
<AnimatedComponent>
  <TouchableHighlight
    activeOpacity={0.85}
    delayPressOut={100}
    onLongPress={[Function]}
    onPress={[Function]}
    underlayColor="rgba(69,120,191,0.5)"
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "row",
            "height": 44,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "paddingLeft": 16,
            },
            undefined,
          ]
        }
      >
        <ChannelIcon
          channelId="channel_id"
          hasDraft={false}
          isActive={false}
          isArchived={true}
          isBot={false}
          isInfo={false}
          isUnread={true}
          membersCount={1}
          size={16}
          status="online"
          theme={
            Object {
              "awayIndicator": "#ffbc42",
              "buttonBg": "#166de0",
              "buttonColor": "#ffffff",
              "centerChannelBg": "#ffffff",
              "centerChannelColor": "#3d3c40",
              "codeTheme": "github",
              "dndIndicator": "#f74343",
              "errorTextColor": "#fd5960",
              "linkColor": "#2389d7",
              "mentionBj": "#ffffff",
              "mentionColor": "#145dbf",
              "mentionHighlightBg": "#ffe577",
              "mentionHighlightLink": "#166de0",
              "newMessageSeparator": "#ff8800",
              "onlineIndicator": "#06d6a0",
              "sidebarBg": "#145dbf",
              "sidebarHeaderBg": "#1153ab",
              "sidebarHeaderTextColor": "#ffffff",
              "sidebarText": "#ffffff",
              "sidebarTextActiveBorder": "#579eff",
              "sidebarTextActiveColor": "#ffffff",
              "sidebarTextHoverBg": "#4578bf",
              "sidebarUnreadText": "#ffffff",
              "type": "Mattermost",
            }
          }
          type="D"
        />
        <Text
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "rgba(255,255,255,0.4)",
                "flex": 1,
                "fontSize": 14,
                "fontWeight": "600",
                "height": "100%",
                "lineHeight": 44,
                "paddingRight": 10,
                "textAlignVertical": "center",
              },
              Object {
                "color": "#ffffff",
              },
            ]
          }
        >
          display_name
        </Text>
      </View>
    </View>
  </TouchableHighlight>
</AnimatedComponent>
`;

exports[`ChannelItem should match snapshot for deactivated user and not searchResults or currentChannel 1`] = `null`;

exports[`ChannelItem should match snapshot for no displayName 1`] = `null`;

exports[`ChannelItem should match snapshot for showUnreadForMsgs 1`] = `null`;

exports[`ChannelItem should match snapshot with draft 1`] = `
<AnimatedComponent>
  <TouchableHighlight
    activeOpacity={0.85}
    delayPressOut={100}
    onLongPress={[Function]}
    onPress={[Function]}
    underlayColor="rgba(69,120,191,0.5)"
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "row",
            "height": 44,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "paddingLeft": 16,
            },
            undefined,
          ]
        }
      >
        <ChannelIcon
          channelId="channel_id"
          hasDraft={true}
          isActive={false}
          isArchived={false}
          isBot={false}
          isInfo={false}
          isUnread={true}
          membersCount={1}
          size={16}
          status="online"
          theme={
            Object {
              "awayIndicator": "#ffbc42",
              "buttonBg": "#166de0",
              "buttonColor": "#ffffff",
              "centerChannelBg": "#ffffff",
              "centerChannelColor": "#3d3c40",
              "codeTheme": "github",
              "dndIndicator": "#f74343",
              "errorTextColor": "#fd5960",
              "linkColor": "#2389d7",
              "mentionBj": "#ffffff",
              "mentionColor": "#145dbf",
              "mentionHighlightBg": "#ffe577",
              "mentionHighlightLink": "#166de0",
              "newMessageSeparator": "#ff8800",
              "onlineIndicator": "#06d6a0",
              "sidebarBg": "#145dbf",
              "sidebarHeaderBg": "#1153ab",
              "sidebarHeaderTextColor": "#ffffff",
              "sidebarText": "#ffffff",
              "sidebarTextActiveBorder": "#579eff",
              "sidebarTextActiveColor": "#ffffff",
              "sidebarTextHoverBg": "#4578bf",
              "sidebarUnreadText": "#ffffff",
              "type": "Mattermost",
            }
          }
          type="O"
        />
        <Text
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "rgba(255,255,255,0.4)",
                "flex": 1,
                "fontSize": 14,
                "fontWeight": "600",
                "height": "100%",
                "lineHeight": 44,
                "paddingRight": 10,
                "textAlignVertical": "center",
              },
              Object {
                "color": "#ffffff",
              },
            ]
          }
        >
          display_name
        </Text>
      </View>
    </View>
  </TouchableHighlight>
</AnimatedComponent>
`;

exports[`ChannelItem should match snapshot with mentions and muted 1`] = `
<AnimatedComponent>
  <TouchableHighlight
    activeOpacity={0.85}
    delayPressOut={100}
    onLongPress={[Function]}
    onPress={[Function]}
    underlayColor="rgba(69,120,191,0.5)"
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "row",
            "height": 44,
          },
          Object {
            "opacity": 0.5,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "paddingLeft": 16,
            },
            undefined,
          ]
        }
      >
        <ChannelIcon
          channelId="channel_id"
          hasDraft={false}
          isActive={false}
          isArchived={false}
          isBot={false}
          isInfo={false}
          isUnread={true}
          membersCount={1}
          size={16}
          status="online"
          theme={
            Object {
              "awayIndicator": "#ffbc42",
              "buttonBg": "#166de0",
              "buttonColor": "#ffffff",
              "centerChannelBg": "#ffffff",
              "centerChannelColor": "#3d3c40",
              "codeTheme": "github",
              "dndIndicator": "#f74343",
              "errorTextColor": "#fd5960",
              "linkColor": "#2389d7",
              "mentionBj": "#ffffff",
              "mentionColor": "#145dbf",
              "mentionHighlightBg": "#ffe577",
              "mentionHighlightLink": "#166de0",
              "newMessageSeparator": "#ff8800",
              "onlineIndicator": "#06d6a0",
              "sidebarBg": "#145dbf",
              "sidebarHeaderBg": "#1153ab",
              "sidebarHeaderTextColor": "#ffffff",
              "sidebarText": "#ffffff",
              "sidebarTextActiveBorder": "#579eff",
              "sidebarTextActiveColor": "#ffffff",
              "sidebarTextHoverBg": "#4578bf",
              "sidebarUnreadText": "#ffffff",
              "type": "Mattermost",
            }
          }
          type="O"
        />
        <Text
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "rgba(255,255,255,0.4)",
                "flex": 1,
                "fontSize": 14,
                "fontWeight": "600",
                "height": "100%",
                "lineHeight": 44,
                "paddingRight": 10,
                "textAlignVertical": "center",
              },
              Object {
                "color": "#ffffff",
              },
            ]
          }
        >
          display_name
        </Text>
        <Badge
          count={1}
          countStyle={
            Object {
              "color": "#145dbf",
              "fontSize": 10,
            }
          }
          extraPaddingHorizontal={10}
          minHeight={20}
          minWidth={20}
          onPress={[Function]}
          style={
            Object {
              "backgroundColor": undefined,
              "borderColor": "#1153ab",
              "borderRadius": 10,
              "borderWidth": 1,
              "padding": 3,
              "position": "relative",
              "right": 16,
            }
          }
        />
      </View>
    </View>
  </TouchableHighlight>
</AnimatedComponent>
`;
